### PR TITLE
Remove indexing code that trying to process all annotations in one go

### PR DESCRIPTION
**Pull request by marcospri**
Found `column_windows` while trying to upgrade SQLAlchemy. It looks like its only usage is a version of indexing that tries to process all annotations in one go without using the job queue.

This seems like an approach that won't work now with and that probably predates the job queue.

Removed both this indexing approach and the `column_windows` function.

### CODE REVIEW

File: h/cli/commands/search.py [hunk 1]
```diff
@@ -11,27 +8,6 @@ def search():
     """Manage search index."""
 
 
-@search.command()
-@click.pass_context
-def reindex(ctx):
```
Comment (marcospri) [inline h/cli/commands/search.py:16]: I don't think we need this or that it would work now.

File: h/indexer/reindexer.py [hunk 2]
```diff
@@ -1,55 +0,0 @@
...
-    try:
-        settings.put(setting_name, new_index)
-        request.tm.commit()
-
-        log.info("reindexing annotations into new index %s", new_index)
-        indexer = BatchIndexer(
-            session, es, request, target_index=new_index, op_type="create"
-        )
-
-        errored = indexer.index()
```
Comment (marcospri) [inline h/indexer/reindexer.py:40]: Calling indexer.index with no params.

File: h/search/index.py [hunk 3]
```diff
@@ -33,24 +32,17 @@ def __init__(  # pylint: disable=too-many-arguments
...
         :param annotation_ids: a list of ids to reindex, reindexes all when `None`.
-        :type annotation_ids: collection
         :param windowsize: the number of annotations to index in between progress log statements
-        :type windowsize: integer
 
         :returns: a set of errored ids
         :rtype: set
         """
-        if annotation_ids is None:
-            annotations = _all_annotations(session=self.session, windowsize=windowsize)
```
Comment (marcospri) [inline h/search/index.py:49]: When no annos are passed, it gets them all from the DB.

This tries to be very efficient but I don't think is a feasible approach anymore.

File: h/search/index.py [hunk 4]
```diff
@@ -89,23 +81,6 @@ def _prepare(self, annotation):
         return {self.op_type: operation}, data
 
 
-def _all_annotations(session, windowsize=2000):
-    # This is using a windowed query for loading all annotations in batches.
-    # It is the most performant way of loading a big set of records from
-    # the database while still supporting eagerloading of associated
-    # document data.
-    windows = column_windows(
```
Comment (marcospri) [inline h/search/index.py:97]: This happens to be the only usage of `column_windows` which gets in the way of upgrading SQLA.

File: h/cli/commands/search.py (hunk 1 above)
Comment (seanh) [reply h/cli/commands/search.py:16]: :+1:

File: h/search/index.py (hunk 3 above)
Comment (seanh) [reply h/search/index.py:49]: What even calls this `BatchIndexer` nowadays? Whatever it is should probably use the job queue and then we could get rid of this entirely

**Review by seanh — APPROVED** [review]

**Pull request merged**